eserde_derive/src/model.rs

```diff
@@ -122,21 +122,52 @@ impl PermissiveCompanionType {
                 if is_eserde_compatible {
                     // Add or replace `#[serde(deserialize_with = "..")]` for our wrapper.
 
-                    // Handle user `#[serde(deserialize_with)]` attributes.
+                    // Handle user `#[serde(deserialize_with = "..")]` or `#[serde(with = "..')]` attributes.
                     let dewith_path =
+                        // Remove `#[serde(deserialize_with = "..")]` and get the string value.
                         remove_attr_meta(&mut field.attrs, "serde", "deserialize_with")
                             .and_then(|meta_item| meta_item.value)
                             .and_then(|(_eq, expr)| {
                                 // Get `expr` as a string literal and parse as a path.
-                                let syn::Expr::Lit(syn::ExprLit {
+                                if let syn::Expr::Lit(syn::ExprLit {
                                     attrs: _,
                                     lit: syn::Lit::Str(lit_str),
                                 }) = expr
-                                else {
-                                    return None;
-                                };
-                                syn::parse_str::<syn::Path>(lit_str.value().as_str()).ok()
-                            });
+                                {
+                                    Some(lit_str.value())
+                                } else {
+                                    None
+                                }
+                            })
+                            // Or else remove `#[serde(with = "..")]` and get the string value.
+                            .or_else(|| remove_attr_meta(&mut field.attrs, "serde", "with")
+                                .and_then(|meta_item| meta_item.value)
+                                .and_then(|(_eq, expr)| {
+                                    // Get `expr` as the string literal value.
+                                    if let syn::Expr::Lit(syn::ExprLit {
+                                        attrs: _,
+                                        lit: syn::Lit::Str(lit_str),
+                                    }) = expr
+                                    {
+                                        let path_str = lit_str.value();
+
+                                        // Desugar the serialize side into `#[serde(serialize_with = "$path_str::serialize")]`.
+                                        let path_ser = syn::LitStr::new(
+                                            &format!("{}::serialize", path_str),
+                                            span,
+                                        );
+                                        field.attrs.push(syn::parse_quote_spanned!(span=>
+                                            #[serde(serialize_with = #path_ser)]
+                                        ));
```

Comment:
We could avoid this if we take care of removing `serde::Serialize` on the type itself. In the end we only care about the deserialization side of things. The less code we generate, the better.

Reply:
Good point, I forgot that this isn't the original struct

Reply:
No matter how I arranged the `#[derive(...)]` inputs `#[derive(serde::Serialize)]` was never included in the macro output. Seems that derive attributes are not included in derives' inputs?

Updated to remove this unused code and cleaned up via helper function